pom.xml

```diff
@@ -235,6 +235,40 @@
                 </plugins>
             </build>
         </profile>
+
+        <profile>
+            <id>java11</id>
+            <activation>
+                <jdk>[11,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>build-helper-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>add-source-Java_Platform_Module_System</id>
+                                <phase>generate-sources</phase>
+                                <goals>
+                                    <goal>add-source</goal>
+                                </goals>
+                                <configuration>
+                                    <sources>
+                                        <source>${project.basedir}/src/main/java11</source>
+                                    </sources>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+            <properties>
+                <project.custom.java.version>11</project.custom.java.version>
+                <maven.compiler.release>11</maven.compiler.release>
+            </properties>
```

Comment:
Note to self: This could be a `<configuration>` section with a `<release>`  element. 
Sort out what `project.custom.java.version` does.